lib/_stream_writable.js

```diff
@@ -107,6 +112,13 @@ function WritableState(options, stream) {
 
   // True if the error was already emitted and should not be thrown again
   this.errorEmitted = false;
+
+  // when `state._end()` was called
+  this.flushed = false;
+
+  // NOTE: added here to not pollute the prototype of the WritableStream and to
+  // avoid conflicts with user-land methods
+  this._end = options._end || _end;
```

Comment:
@indutny my suggestion was on the `writeableState.prototype` rather than explicitly assigning upon construction. But it is a little bit 'tit for tat' as the performance gained from the use of the prototype is probably minimal :smile: